sisyphus/manager.py

```diff
@@ -572,6 +576,12 @@ def maybe_clear_state(state, always_clear, action):
             self.job_cleaner.start()
         return True
 
+    def handle_job_failure(self, prev_jobs: Dict[str, List[Job]], cur_jobs: Dict[str, List[Job]]):
+        prev_errored_jobs = set(prev_jobs.get(gs.STATE_ERROR, []))
+        for job in cur_jobs.get(gs.STATE_ERROR, []):
+            if job not in prev_errored_jobs:
+                gs.on_job_failure(job)
+
```

Comment:
would you call `self.update_jobs()` after finishing the loop to account for changes in the state of jobs?

Reply:
Not needed, since this will loop around immediately when the processing is done.